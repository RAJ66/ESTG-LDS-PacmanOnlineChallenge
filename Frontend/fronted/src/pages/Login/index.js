import React, { useState } from "react";
import api from "../../services/api";
import logo from "../../assets/logo.png";

export default function Login({ history }) {
  const [email, setEmail] = useState("");
  const [pass, setPass] = useState("");

  function handleSubmit(event) {
    event.preventDefault();
    console.log(email);
    console.log(pass);

    api.post("/api/users", {
      UserEmail: email,
      Password: pass
    });

    history.push("/dashboard");
  }

  return (
    <form className="login-container" onSubmit={handleSubmit}>
      <img src={logo} alt="It a match" />
      <input
        placeholder="Email"
        value={email}
        onChange={event => setEmail(event.target.value)}
      />
      <input
        placeholder="Password"
        type="password"
        value={pass}
        onChange={event => setPass(event.target.value)}
      />

      <button type="submit" className="glow-on-hover">
        Login
      </button>
      <button type="" className="glow-on-hover">
        Register
      </button>
    </form>
  );
}
