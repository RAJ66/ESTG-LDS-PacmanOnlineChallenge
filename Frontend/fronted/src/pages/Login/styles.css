.container-inicio {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background: var(--color-base);
}

.login-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  max-width: 400px;
  width: 100%;

  background: var(--color-zero);
  border-radius: 6px;
  padding: 30px;
}

.login-container input {
  margin-top: 20px;
  border: 2px solid var(--color-base);
  border-radius: 4px;
  height: 48px;
  padding: 0 20px;
  font-size: 16px;
  color: var(--color-base);
}

.login-container input::placeholder {
  color: var(--color-base);
}

.login-container button {
  margin-top: 10px;
  border: 0px;
  border-radius: 4px;
  height: 46px;
  width: 100px;
  font-size: 16px;
  font-weight: bold;
  color: var(--color-zero);
  cursor: pointer;
}

.glow-on-hover {
  width: 220px;
  height: 50px;
  border: none;
  outline: none;
  color: var(--color-zero);
  background: var(--color-base);
  cursor: pointer;
  position: relative;
  z-index: 0;
  border-radius: 10px;
}

.glow-on-hover:before {
  content: "";
  background: linear-gradient(
    45deg,
    #ff0000,
    #ff7300,
    #fffb00,
    #48ff00,
    #00ffd5,
    #002bff,
    #7a00ff,
    #ff00c8,
    #ff0000
  );
  position: absolute;
  top: -2px;
  left: -2px;
  background-size: 400%;
  z-index: -1;
  filter: blur(5px);
  width: calc(100% + 4px);
  height: calc(100% + 4px);
  animation: glowing 20s linear infinite;
  opacity: 0;
  transition: opacity 0.3s ease-in-out;
  border-radius: 10px;
}

.glow-on-hover:active {
  color: var(--color-base);
}

.glow-on-hover:active:after {
  background: transparent;
}

.glow-on-hover:hover:before {
  opacity: 1;
}

.glow-on-hover:after {
  z-index: -1;
  content: "";
  position: absolute;
  width: 100%;
  height: 100%;
  background: var(--color-base);
  left: 0;
  top: 0;
  border-radius: 10px;
}

@keyframes glowing {
  0% {
    background-position: 0 0;
  }
  50% {
    background-position: 400% 0;
  }
  100% {
    background-position: 0 0;
  }
}

.login-container img {
  height: 100px;
  width: 150px;
}

@media only screen and (max-width: 600px) {
  body,
  button {
    background: var(--color-base);
    -webkit-font-smoothing: antialiased !important;
  }

  .container {
    width: 100vw;
    height: 100vh;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    background: var(--color-base);
  }

  .login-container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border-radius: 0px;
    background: var(--color-zero);
    max-width: 600px;
    width: 100%;
    padding: 30px;
  }

  .login-container input {
    margin-top: 20px;
    border: 2px solid var(--color-base);
    border-radius: 4px;
    height: 46px;
    padding: 0 20px;
    font-size: 16px;
    color: var(--color-base);
  }

  .login-container input::placeholder {
    color: var(--color-base);
  }

  .login-container button {
    margin-top: 10px;
    border: 0px;
    border-radius: 4px;
    height: 46px;
    width: 100px;
    font-size: 16px;
    font-weight: bold;
    color: var(--color-zero);
    cursor: pointer;
  }
}
