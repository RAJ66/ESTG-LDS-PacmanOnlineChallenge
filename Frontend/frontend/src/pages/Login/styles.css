.container-inicio {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background: var(--color-base);
}

.login-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  max-width: 400px;
  width: 100%;

  background: var(--color-zero);
  border-radius: 6px;
  padding: 30px;
}

.login-container input {
  margin-top: var(--size-small-medium);
  border: 2px solid var(--color-base);
  border-radius: 4px;
  height: 48px;
  padding: 0 var(--size-small-medium);
  font-size: 16px;
  color: var(--color-base);
}

.login-container button {
  margin-top: var(--size-small);
  border: 0px;
  border-radius: 4px;
  height: var(--size-medium);
  width: var(--size-big);
  font-size: 16px;
  font-weight: bold;
  color: var(--color-zero);
  cursor: pointer;
}

.glow-on-hover {
  width: 220px;
  height: 50px;
  border: none;
  outline: none;
  color: var(--color-zero);
  background: var(--color-base);
  cursor: pointer;
  position: relative;
  z-index: 0;
  border-radius: var(--size-small);
}

.glow-on-hover:before {
  content: "";
  background: linear-gradient(
    45deg,
    #ff0000,
    #ff7300,
    #fffb00,
    #48ff00,
    #00ffd5,
    #002bff,
    #7a00ff,
    #ff00c8,
    #ff0000
  );
  position: absolute;
  top: -2px;
  left: -2px;
  background-size: 400%;
  z-index: -1;
  filter: blur(5px);
  width: calc(100% + 4px);
  height: calc(100% + 4px);
  animation: glowing 20s linear infinite;
  opacity: 0;
  transition: opacity 0.3s ease-in-out;
  border-radius: var(--size-small);
}

.glow-on-hover:active {
  color: var(--color-base);
}

.glow-on-hover:active:after {
  background: transparent;
}

.glow-on-hover:hover:before {
  opacity: 1;
}

.glow-on-hover:after {
  z-index: -1;
  content: "";
  position: absolute;
  width: 100%;
  height: 100%;
  background: var(--color-base);
  left: 0;
  top: 0;
  border-radius: var(--size-small);
}

@keyframes glowing {
  0% {
    background-position: 0 0;
  }
  50% {
    background-position: 400% 0;
  }
  100% {
    background-position: 0 0;
  }
}

.logo {
  height: var(--size-big);
  width: 150px;
}

@media only screen and (max-width: 600px) {
  .login-container {
    border-radius: 0px;
  }
}

@media only screen and (max-height: 550px) {
  .logo {
    height: 75px;
    width: 125px;
  }
  .container-login {
    width: 100vw;
    height: 100vh;
  }

  .login-container {
    width: 100vw;
    height: 100vh;
  }

  .login-container {
    border-radius: 0px;
  }
}
