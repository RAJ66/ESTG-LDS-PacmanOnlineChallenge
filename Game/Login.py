import pygame
import json
import requests
from MenuPrincipal import menuPrincipal

pygame.init()
screen = pygame.display.set_mode((610, 670))
COLOR_INACTIVE = pygame.Color('lightskyblue3')
COLOR_ACTIVE = pygame.Color('dodgerblue2')
FONT = pygame.font.Font(None, 32)


class InputBox:
    def __init__(self, x, y, w, h, n, text=''):
        self.rect = pygame.Rect(x, y, w, h)
        self.color = COLOR_INACTIVE
        self.text = text
        self.txt_surface = FONT.render(text, True, self.color)
        self.active = False
        self.font = pygame.font.Font('PAC-FONT.TTF', 25)
        self.hide = ''
        self.number = n

    def handle_event(self, event):
        if event.type == pygame.MOUSEBUTTONDOWN:
            if self.rect.collidepoint(event.pos):
                self.active = not self.active
            else:
                self.active = False
            self.color = COLOR_ACTIVE if self.active else COLOR_INACTIVE
        if event.type == pygame.KEYDOWN:
            password = ''
            if self.active:
                if event.key == pygame.K_RETURN:
                    self.text = ''
                elif event.key == pygame.K_BACKSPACE:
                    self.text = self.text[:-1]
                    self.hide = self.hide[:-1]
                else:
                    password += '*'
                    self.text += event.unicode
                    self.hide += '*'
                if self.number == 1:
                    self.txt_surface = FONT.render(self.text, True, self.color)
                else:
                    self.txt_surface = FONT.render(self.hide, True, self.color)

    def update(self):
        width = max(200, self.txt_surface.get_width()+10)
        self.rect.w = width

    def draw(self, screen):
        screen.blit(self.txt_surface, (self.rect.x+5, self.rect.y+5))
        pygame.draw.rect(screen, self.color, self.rect, 2)

    def addText(self):
        screen.blit(pygame.font.Font('PAC-FONT.TTF', 25).render(
            'login', True, (255, 0, 0)), (260, 400))

    def addText2(self):
        screen.blit(pygame.font.Font('PAC-FONT.TTF', 25).render(
            'ONLINE CHALLENGE', True, (255, 255, 0)), (150, 250))

    def addText3(self):
        screen.blit(pygame.font.Font('PAC-FONT.TTF', 15).render(
            'CLICK HERE IF YOU FORGOT YOUR ACCOUNT DETAILS', True, (255, 0, 0)), (20, 600))


def main():
    clock = pygame.time.Clock()
    input_box1 = InputBox(220, 300, 140, 32, 1, "Username")
    input_box2 = InputBox(220, 350, 140, 32, 2, "")
    input_boxes = [input_box1, input_box2]
    done = False

    username = None
    password = None

    while not done:

        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                done = True
            if event.type == pygame.KEYDOWN:
                if event.key == pygame.K_RETURN:
                    url = "https://localhost:5001/api/user/auth"
                    data = {'Username': username, 'Password': password}
                    headers = {'Content-type': 'application/json',
                               'Accept': 'text/plain'}
                    r = requests.post(url, data=json.dumps(
                        data), headers=headers, verify=False)
                    if r.status_code == 200:
                        menuPrincipal(username)
                    else:
                        print("USERNAME OU PASSWORD INVALIDA")
                        main()
            for box in input_boxes:
                box.handle_event(event)

        for box in input_boxes:
            box.update()

        screen.fill((0, 0, 0))
        for box in input_boxes:
            box.draw(screen)

        print(username)
        print(password)

        username = input_box1.text.strip()
        password = input_box2.text.strip()

        InputBox.addText3(screen)
        InputBox.addText2(screen)
        button = pygame.Rect(260, 400, 100, 40)
        pygame.draw.rect(screen, [0, 0, 0], button)
        InputBox.addText(screen)
        img = pygame.image.load('img/logo.jpg')
        screen.blit(pygame.transform.scale(img, (610, 250)), (0, 0))
        pygame.display.flip()
        clock.tick(10)


main()
pygame.quit()
