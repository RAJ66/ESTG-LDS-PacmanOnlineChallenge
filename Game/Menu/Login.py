import pygame


def main():
    WHITE = (255,255,255)
    YELLOW = (173,255,47)
    screen = pygame.display.set_mode((640, 480))
    font = pygame.font.Font(None, 32)
    clock = pygame.time.Clock()
    input_box = pygame.Rect(100, 100, 140, 32)
    color_inactive = pygame.Color(WHITE)
    color_active = pygame.Color(YELLOW)
    color = color_inactive
    active = False
    boxPass = ''
    password = ''
    done = False

    while not done:
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                done = True
            if event.type == pygame.MOUSEBUTTONDOWN:
                if input_box.collidepoint(event.pos):
                    active = not active
                else:
                    active = False
                color = color_active if active else color_inactive
            if event.type == pygame.KEYDOWN:
                if active:
                    if event.key == pygame.K_RETURN:
                        print(password)
                        boxPass = ''
                    elif event.key == pygame.K_BACKSPACE:
                        boxPass = boxPass[:-1]
                        password = password[:-1]
                    else:
                        password += event.unicode
                        boxPass += '*'

        screen.fill((30, 30, 30))

        txt_surface = font.render(boxPass, True, color)

        width = max(200, txt_surface.get_width()+10)
        input_box.w = width

        screen.blit(txt_surface, (input_box.x+5, input_box.y+5))

        pygame.draw.rect(screen, color, input_box, 2)

        pygame.display.flip()
        clock.tick(30)


if __name__ == '__main__':
    pygame.init()
    main()
    pygame.quit()