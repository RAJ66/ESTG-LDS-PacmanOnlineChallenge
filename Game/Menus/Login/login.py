


from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_MainWindows(object):
    def setupUi(self, MainWindows):
        MainWindows.setObjectName("MainWindows")
        MainWindows.resize(637, 557)
        self.centralwidget = QtWidgets.QWidget(MainWindows)
        self.centralwidget.setObjectName("centralwidget")
        self.btn_Logar = QtWidgets.QPushButton(self.centralwidget)
        self.btn_Logar.setGeometry(QtCore.QRect(240, 420, 151, 51))
        self.btn_Logar.setObjectName("btn_Logar")
        self.textEdit = QtWidgets.QTextEdit(self.centralwidget)
        self.textEdit.setGeometry(QtCore.QRect(160, 280, 421, 41))
        font = QtGui.QFont()
        font.setPointSize(19)
        self.textEdit.setFont(font)
        self.textEdit.setObjectName("textEdit")
        self.textEdit_2 = QtWidgets.QTextEdit(self.centralwidget)
        self.textEdit_2.setGeometry(QtCore.QRect(160, 350, 421, 41))
        font = QtGui.QFont()
        font.setPointSize(19)
        self.textEdit_2.setFont(font)
        self.textEdit_2.setObjectName("textEdit_2")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(70, 290, 81, 41))
        font = QtGui.QFont()
        font.setPointSize(19)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_2.setGeometry(QtCore.QRect(20, 340, 131, 61))
        font = QtGui.QFont()
        font.setPointSize(19)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.label_3 = QtWidgets.QLabel(self.centralwidget)
        self.label_3.setGeometry(QtCore.QRect(370, 490, 171, 51))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")
        self.pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton.setGeometry(QtCore.QRect(549, 500, 81, 28))
        self.pushButton.setObjectName("pushButton")
        self.label_4 = QtWidgets.QLabel(self.centralwidget)
        self.label_4.setGeometry(QtCore.QRect(160, 0, 331, 271))
        self.label_4.setText("")
        self.label_4.setPixmap(QtGui.QPixmap("image.png"))
        self.label_4.setObjectName("label_4")
        MainWindows.setCentralWidget(self.centralwidget)
        self.statusbar = QtWidgets.QStatusBar(MainWindows)
        self.statusbar.setObjectName("statusbar")
        MainWindows.setStatusBar(self.statusbar)

        self.retranslateUi(MainWindows)
        QtCore.QMetaObject.connectSlotsByName(MainWindows)

    def retranslateUi(self, MainWindows):
        _translate = QtCore.QCoreApplication.translate
        MainWindows.setWindowTitle(_translate("MainWindows", "POC - Pacman Online Challenge"))
        self.btn_Logar.setText(_translate("MainWindows", "Login"))
        self.label.setText(_translate("MainWindows", "Email:"))
        self.label_2.setText(_translate("MainWindows", "Password:"))
        self.label_3.setText(_translate("MainWindows", "Don\'t have account yet?"))
        self.pushButton.setText(_translate("MainWindows", "Register"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    MainWindows = QtWidgets.QMainWindow()
    ui = Ui_MainWindows()
    ui.setupUi(MainWindows)
    MainWindows.show()
    sys.exit(app.exec_())
