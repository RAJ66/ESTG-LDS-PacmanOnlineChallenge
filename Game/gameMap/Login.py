import pygame as pg

pg.init()
screen = pg.display.set_mode((640, 480))
COLOR_INACTIVE = pg.Color('lightskyblue3')
COLOR_ACTIVE = pg.Color('dodgerblue2')
FONT = pg.font.Font(None, 32)


class InputBox:
    def __init__(self, x, y, w, h, text=''):
        self.rect = pg.Rect(x, y, w, h)
        self.color = COLOR_INACTIVE
        self.text = text
        self.txt_surface = FONT.render(text, True, self.color)
        self.active = False
        self.font = pg.font.SysFont('Arial', 25)

    def handle_event(self, event):
        if event.type == pg.MOUSEBUTTONDOWN:
            if self.rect.collidepoint(event.pos):
                self.active = not self.active
            else:
                self.active = False
            self.color = COLOR_ACTIVE if self.active else COLOR_INACTIVE
        if event.type == pg.KEYDOWN:
            password = ''
            if self.active:
                if event.key == pg.K_RETURN:
                    self.text = ''
                    print("123")
                    print(password)
                elif event.key == pg.K_BACKSPACE:
                    self.text = self.text[:-1]
                else:
                    password += '*'
                    self.text += event.unicode
                self.txt_surface = FONT.render(self.text, True, self.color)

    def update(self):
        width = max(200, self.txt_surface.get_width()+10)
        self.rect.w = width

    def draw(self, screen):
        screen.blit(self.txt_surface, (self.rect.x+5, self.rect.y+5))
        pg.draw.rect(screen, self.color, self.rect, 2)

    def addText(self):
        screen.blit(pg.font.SysFont('PacFont', 25).render(
            'login', True, (255, 0, 0)), (260, 300))


def main():
    clock = pg.time.Clock()
    input_box1 = InputBox(200, 200, 140, 32)
    input_box2 = InputBox(200, 250, 140, 32)
    input_boxes = [input_box1, input_box2]
    done = False

    while not done:
        for event in pg.event.get():
            if event.type == pg.QUIT:
                done = True
            for box in input_boxes:
                box.handle_event(event)

        for box in input_boxes:
            box.update()

        screen.fill((30, 30, 30))
        for box in input_boxes:
            box.draw(screen)

        button = pg.Rect(260, 300, 100, 40)
        pg.draw.rect(screen, [255, 255, 255], button)
        InputBox.addText(screen)
        img = pg.image.load('img/logo.png')
        screen.blit(pg.transform.scale(img, (200, 150)), (200, 10))
        pg.display.flip()
        clock.tick(30)


if __name__ == '__main__':
    main()
    pg.quit()
