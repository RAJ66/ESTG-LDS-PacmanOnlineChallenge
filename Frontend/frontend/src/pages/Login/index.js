import React, { useState } from "react";
import api from "../../services/api";
import logo from "../../assets/logo.png";
import "./styles.css";

export default function Login({ history }) {
  const [email, setEmail] = useState("");
  const [pass, setPass] = useState("");

  function handleSubmit(event) {
    event.preventDefault();
    console.log(email);
    console.log(pass);

    api.post("/api/user/auth", {
      username: email,
      password: pass
    });
  }

  function handleClickRegister(event) {
    event.preventDefault();
    history.push("/register");
  }

  function handleClickForgot(event) {
    event.preventDefault();
    alert("Email send");
    history.push("/");
  }

  return (
    <div className="container-inicio">
      <form className="login-container" onSubmit={handleSubmit}>
        <img src={logo} className="logo" alt="It a match" />
        <input
          placeholder="Email"
          value={email}
          onChange={event => setEmail(event.target.value)}
        />
        <input
          placeholder="Password"
          type="password"
          value={pass}
          onChange={event => setPass(event.target.value)}
        />

        <button type="submit" className="glow-on-hover basebutton">
          Login
        </button>
        <button
          type="reset"
          onClick={handleClickRegister}
          className="glow-on-hover basebutton"
        >
          Register
        </button>

        <button onClick={handleClickForgot} className="forgotButton ">
          Forgot account?
        </button>
      </form>
    </div>
  );
}
