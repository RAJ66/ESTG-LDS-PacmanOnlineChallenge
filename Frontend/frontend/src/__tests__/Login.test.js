import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Login from "./../pages/Login/index";

describe("Testing Pacman Component", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Login />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
