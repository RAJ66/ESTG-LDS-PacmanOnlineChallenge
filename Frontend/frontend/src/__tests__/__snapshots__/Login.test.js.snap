// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pacman Component should render correctly 1`] = `
<Login>
  <div
    className="container-inicio"
  >
    <form
      className="login-container"
      onSubmit={[Function]}
    >
      <img
        alt="It a match"
        className="logo"
        src="logo.png"
      />
      <input
        onChange={[Function]}
        placeholder="Username"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <button
        className="glow-on-hover basebutton"
        type="submit"
      >
        Login
      </button>
      <button
        className="glow-on-hover basebutton"
        onClick={[Function]}
        type="reset"
      >
        Register
      </button>
      <button
        className="forgotButton "
        onClick={[Function]}
      >
        Forgot account?
      </button>
    </form>
  </div>
</Login>
`;
